Sources/SuperwallKit/Superwall.swift

```diff
@@ -0,0 +1,354 @@
+  ///   - options: A ``SuperwallOptions`` object which allows you to customise the appearance and behavior of the paywall.
+  /// - Returns: The newly configured ``Superwall`` instance.
+  @discardableResult
+  public static func configure(
+    apiKey: String,
+    delegate: SuperwallDelegate? = nil,
+    options: SuperwallOptions? = nil
+  ) -> Superwall {
+    guard superwall == nil else {
+      Logger.debug(
+        logLevel: .warn,
+        scope: .superwallCore,
+        message: "Superwall.configure called multiple times. Please make sure you only call this once on app launch."
+      )
+      return shared
+    }
+    superwall = Superwall(
+      apiKey: apiKey,
+      swiftDelegate: delegate,
+      objcDelegate: nil,
+      options: options
+    )
+    return shared
+  }
+
+  /// Objective-C only function that configures a shared instance of ``SuperwallKit/Superwall`` for use throughout your app.
+  ///
+  /// Call this as soon as your app finishes launching in `application(_:didFinishLaunchingWithOptions:)`. Check out our <doc:GettingStarted> article for a tutorial on how to configure the SDK.
+  /// - Parameters:
+  ///   - apiKey: Your Public API Key that you can get from the Superwall dashboard settings. If you don't have an account, you can [sign up for free](https://superwall.com/sign-up).
+  ///   - delegate: An optional class that conforms to ``SuperwallDelegate``. The delegate methods receive callbacks from the SDK in response to certain events on the paywall.
+  ///   - options: A ``SuperwallOptions`` object which allows you to customise the appearance and behavior of the paywall.
+  /// - Returns: The newly configured ``SuperwallKit/Superwall`` instance.
+  @discardableResult
+  @available(swift, obsoleted: 1.0)
+  public static func configure(
+    apiKey: String,
+    delegate: SuperwallDelegateObjc? = nil,
+    options: SuperwallOptions? = nil
+  ) -> Superwall {
+    guard superwall == nil else {
+      Logger.debug(
+        logLevel: .warn,
+        scope: .superwallCore,
+        message: "Superwall.configure called multiple times. Please make sure you only call this once on app launch."
+      )
+      return shared
+    }
+    superwall = Superwall(
+      apiKey: apiKey,
+      swiftDelegate: nil,
+      objcDelegate: delegate,
+      options: options
+    )
+    return shared
+  }
+
+  // MARK: - Preloading
+
+  /// Preloads all paywalls that the user may see based on campaigns and triggers turned on in your Superwall dashboard.
+  ///
+  /// To use this, first set ``PaywallOptions/shouldPreload``  to `false` when configuring the SDK. Then call this function when you would like preloading to begin.
+  ///
+  /// Note: This will not reload any paywalls you've already preloaded via ``SuperwallKit/Superwall/preloadPaywalls(forEvents:)``.
+  public static func preloadAllPaywalls() {
```

Comment:
Everything was static to start with, I think just because it's simpler to write `Superwall.track` etc. The reason I had to introduce `Superwall.shared` was because you can only listen to `@Published` properties from instance variables. That's so you can listen to `hasActiveSubscription` or `isConfigured`. Agree there's a bit of a disconnect between the two... I'm just concerned that if we move to using instance function it's just extra space taken up by a `.shared` just to appease  `hasActiveSubscription` or `isConfigured` which isn't going to be used nearly as much

Reply:
I agree its somewhat simpler, but I think we should have a consistent / cohesive API that we expose – if the developer is supposed to use the `sharedInstance` then that's what we should enforce. Could possibly make sense to add the static variants to a convenience extension that internally uses the shared instance, but right now it feels too inconsistent.